From 58aba98d66be4024f9da62944b1cd440bdeae35e Mon Sep 17 00:00:00 2001
From: Tom Rondeau <trondeau@vt.edu>
Date: Mon, 16 Jan 2012 10:09:01 -0500
Subject: [PATCH 08/47] digital: fixing up BERT interfaces. Also added a gain
 control to the tx side (so options.amplitude is
 actually used).

---
 gr-digital/examples/narrowband/digital_bert_rx.py |    2 ++
 gr-digital/examples/narrowband/digital_bert_tx.py |    7 +++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gr-digital/examples/narrowband/digital_bert_rx.py b/gr-digital/examples/narrowband/digital_bert_rx.py
index 2833131..dfed0ee 100755
--- a/gr-digital/examples/narrowband/digital_bert_rx.py
+++ b/gr-digital/examples/narrowband/digital_bert_rx.py
@@ -113,9 +113,11 @@ class rx_psk_block(gr.top_block):
 	self._demodulator = self._demodulator_class(**demod_kwargs)
 
         if(options.rx_freq is not None):
+            symbol_rate = options.bitrate / self._demodulator.bits_per_symbol()
             self._source = uhd_receiver(options.args, options.bitrate,
                                         options.samples_per_symbol,
                                         options.rx_freq, options.rx_gain,
+                                        options.spec,
                                         options.antenna, options.verbose)
             options.samples_per_symbol = self._source._sps
 
diff --git a/gr-digital/examples/narrowband/digital_bert_tx.py b/gr-digital/examples/narrowband/digital_bert_tx.py
index 46f4f90..f29e997 100755
--- a/gr-digital/examples/narrowband/digital_bert_tx.py
+++ b/gr-digital/examples/narrowband/digital_bert_tx.py
@@ -67,9 +67,11 @@ class tx_psk_block(gr.top_block):
 	self._modulator = self._modulator_class(**mod_kwargs)
 
         if(options.tx_freq is not None):
-            self._sink = uhd_transmitter(options.args, options.bitrate,
+            symbol_rate = options.bitrate / self._modulator.bits_per_symbol()
+            self._sink = uhd_transmitter(options.args, symbol_rate,
                                          options.samples_per_symbol,
                                          options.tx_freq, options.tx_gain,
+                                         options.spec,
                                          options.antenna, options.verbose)
             options.samples_per_symbol = self._sink._sps
             
@@ -87,7 +89,8 @@ class tx_psk_block(gr.top_block):
                                           verbose=options.verbose,
                                           log=options.log)
 
-	self.connect(self._transmitter, self._sink)
+        self.amp = gr.multiply_const_cc(options.amplitude)
+	self.connect(self._transmitter, self.amp, self._sink)
 
 
 def get_options(mods):
-- 
1.7.9.1

